doc/extensions.rst

```diff
@@ -64,73 +64,78 @@ interface the new extension.
     model into an OpenML Flow. The flow should preserve the class, the library
-    version, and the tunable hyperparameters. 
+    version, and the tunable hyperparameters.
   * :meth:`get_version_information`: Return a tuple with the version information
-    of the important libraries. 
-  * :meth:`create_setup_string`: No longer used, and will be deprecated soon. 
+    of the important libraries.
+  * :meth:`create_setup_string`: No longer used, and will be deprecated soon.
 * Performing runs (required)
-  * :meth:`is_estimator`: Gets as input a class, and checks whether it has the 
+
+  * :meth:`is_estimator`: Gets as input a class, and checks whether it has the
     status of estimator in the library (typically, whether it has a train method
     and a predict method).
-  * :meth:`seed_model`: Sets a random seed to the model. 
+  * :meth:`seed_model`: Sets a random seed to the model.
   * :meth:`_run_model_on_fold`: One of the main requirements for a library to
     generate run objects for the OpenML server. Obtains a train split (with
     labels) and a test split (without labels) and the goal is to train a model
     on the train split and return the predictions on the test split.
     On top of the actual predictions, also the class probabilities should be
-    determined. 
+    determined.
     For classifiers that do not return class probabilities, this can just be the
     hot-encoded predicted label.
     The predictions will be evaluated on the OpenML server.
     Also, additional information can be returned, for example, user-defined
     measures (such as runtime information, as this can not be inferred on the
-    server). 
+    server).
     Additionally, information about a hyperparameter optimization trace can be
-    provided. 
+    provided.
   * :meth:`obtain_parameter_values`: Obtains the hyperparameters of a given
     model and the current values. Please note that in the case of a hyperparameter
     optimization procedure (e.g., random search), you only should return the
     hyperparameters of this procedure (e.g., the hyperparameter grid, budget,
-    etc) and that the chosen model will be inferred from the optimization trace. 
+    etc) and that the chosen model will be inferred from the optimization trace.
   * :meth:`check_if_model_fitted`: Check whether the train method of the model
     has been called (and as such, whether the predict method can be used).
 * Hyperparameter optimization (optional)
+
   * :meth:`instantiate_model_from_hpo_class`: If a given run has recorded the
     hyperparameter optimization trace, then this method can be used to
     reinstantiate the model with hyperparameters of a given hyperparameter
     optimization iteration. Has some similarities with :meth:`flow_to_model` (as
-    this method also sets the hyperparameters of a model). 
-    
-
+    this method also sets the hyperparameters of a model).
+    Note that although this method is optional, for the extension to work it
```

Comment:
Shouldn't it rather be that the implementing the logic is optional but that the function needs to exists? So

```
Note that although this method is required, it is not necessary to implement any logic if hyperparameter optimization is not implemented. Simply raise a `NotImplementedError` then.
```